[React] Add Traffic Light challenge

rishabhm05 opened the pull request:
# Title and Issue number 
<!-- Please make sure issue number is mention in Pull Request else PR will not be merged. -->
Title : TrafficLightChallenge

Issue No. :

Code Stack :  React JS

Close #<issue_no>
<!-- Example Close #244  -->
<!-- Replace `issue_no` with the issue number which is fixed in this PR -->

# Checklist:

- [] I have mentioned the issue number in my Pull Request.
- [] I have commented my code, particularly in hard-to-understand areas
- [] I have created a helpful and easy to understand `README.md`
- [] I have updated the Index.html file for my contribution
<!-- [X] - put a cross/X inside [] to check the box -->

arpansaha13 commented on react/src/helpers/challenges.ts:
```diff
@@ -445,4 +445,14 @@ export const challenges = new Map<string, Challenge>([
       tags: [],
     },
   ],
+  [
+    'traffic-lights',
+    {
+      title: 'Traffic Lights',
+      link: 'traffic-lights',
+      difficulty: 'medium',
+      developer: 'rishabhm05',
+      tags: [],
+    },
+  ],
```
Put the new challenge along with the other medium difficulty challenges

arpansaha13 reviewed (changes requested):
Consider showing a timer for how long the red and green lights will show

arpansaha13 commented on react/src/machine-coding/traffic-light/App.jsx:
```diff
@@ -0,0 +1,30 @@
+  return (
+    <div className={styles.AppContainer}>
+      <div className={styles.trafficWrapper}>
+        {Object.keys(config).map((light, index) => {
+          return (
+            <div
+              className={styles.light}
+              style={{ background: `${currentActiveLight === light ? `${light}` : ''}` }}
+            ></div>
+          );
+        })}
```
Each item in list-rendering should have a unique `key` prop. Please add the `key` prop in the list elements.

Note that the `index` variable is unused. Please remove it if it is not required. Also, ideally, index should not be used as key.

rishabhm05 commented:
@arpansaha13  Hi  i have done required changes
1 . Added key for each light item
2. Displayed duration for Active Light
3 . Added challenge below the other medium challenges

arpansaha13 commented:
Thank you for your contribution!

arpansaha13 commented:
I have noticed a bug right now. Check out this issue #325.
Would you like to take it up?

rishabhm05 commented:
yes @arpansaha13  Please assign it to me.

arpansaha13 commented:
I am not able to add you as an assignee.
But you can open the PR for it anyway 👍

rishabhm05 commented:
okay sure
